imports.go

```diff
@@ -39,80 +39,111 @@ import (
 	"strconv"
 	"strings"
 	"time"
+
+	"github.com/pkg/errors"
 )
 
+type Record interface {
```

Comment:
The `index` is the index of a field in a record, e.g., the CSV line `45,52` is a record with two fields. Index of the first field (45) is 0, and index of the second field (52) is 1. `Uint64Field` is used for row/column IDs and `Int64Field` is used for BSI values. `StringField` is to be used with keys.  `Bit` and `FieldValues` structs implement the `Record` interface in `imports.go`. Since we don't support importing anything other than Bits and Field values in the server, there's not much point in adding more structs which implement the `Record` interface (for now).

Reply:
Here's the doc I've written about imports: https://github.com/yuce/go-pilosa/blob/optimize-imports-refactor-importbits/docs/imports-exports.md

Reply:
This pattern reminds me of the issue I raised in #134; it seems like `Bit` could just implement a basic `Record` interface:
```
type Record interface {
    Less(other Record) bool
}
```
 In that case, the code in `bitsToImportRequest()` would look something like:
```
func bitsToImportRequest(indexName string, frameName string, slice uint64, records []Record) *pbuf.ImportRequest {
    ...
    for _, record := range records {
        bit := record.(Bit)
        rowIDs = append(rowIDs, bit.RowID)
        columnIDs = append(columnIDs, bit.ColumnID)
        timestamps = append(timestamps, bit.Timestamp)
    }
    ...
}
```
And all the `*Field(index int)` methods could go away.

Maybe I'm missing something though. Does the `index` logic provide any benefit?

Reply:
@travisturner What you propose is great and simplifies a great deal of code. I'll see about implementing #134 too. At least the `Uint64Field` and the index logic is required in `import_manager.go` since it is used to get the column ID for both `Bit` and `FieldValue`s at lines: 48, 126. Having a `ColumnID` function in the interface instead of having to use `Uint64Field(1)` (or a function which would return the slice for a `Record`) would be better but we can change that stuff later without affecting any other code.